### ISSUE TITLE
Slow SRP calculations in Node

### ISSUE THREAD

**Issue by thchia** [aws/amazon-cognito-identity-js#368]
In reference to PR #282, which replaces the jsbn dependency with a hard-coded implementation of BigInteger, I am seeing relatively slow authentication times in a node environment (ranging from ~5s - ~10s).

@itrestian mentioned in #43 that using jsbn got his auth times down to ~2s but I'm not seeing this even when I swap out the internal BigInteger implementation for the previous jsbn one.

Just wondering if anyone is using this in node and what your experiences/workarounds are. Thanks.

**Comment by itrestian** [comment]
That is the same implementation that we were previously using. We just pulled it into our code base. We have seen slow performance in hybrid environments, haven't seen anything of that sort in node yet.

**Comment by thchia** [comment]
Ok, thanks for the quick response. I will dig around a bit more to try and find out what the cause is.

**Comment by itrestian** [comment]
If you can pinpoint the exact function that is taking a long time that would be helpful. Is it the mod pow as in other cases?

**Comment by thchia** [comment]
On closer inspection it appears the delays are stemming from the various `makeUnauthenticatedRequest` functions.

I suppose this could be due to the location of the User Pool (us-west-2, I am testing from ap-southeast-1/Singapore). Oddly today I'm getting times consistently around the 4s mark even though I haven't touched anything, which is much better than 10s but not as a good as sub-2s.

Closing since it's not related to SRP (I checked and it's not the modPow functions).
